### PR TITLE
Custom theme refactoring changes

### PR DESCRIPTION
**Pull request by caalador**
Only consider styles.css from the root folder and imports found there.
Fix for static copying resources.
Log level change to be less noisy by default.
fix creation of folders so they are made only when needed.

Part of #9983

### CODE REVIEW

**Review by pleku — COMMENTED** [review]
Just one question

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-copy.js [hunk 1]
```diff
@@ -22,6 +22,51 @@ const fs = require('fs');
 const path = require('path');
 const glob = require('glob');
 
+const ignoredFileExtensions = [".css", ".js", ".json"];
+
+/**
+ * Copy theme static resources to static assets folder. All files in the theme
+ * folder will be copied excluding css, js and json files that will be
```
[the remark this answers is not in the document]
Comment (caalador) [reply flow-server/src/main/resources/plugins/application-theme-plugin/theme-copy.js:30]: `@CssImport` should be bundled by webpack so not copying that shouldn't make a difference.

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-copy.js [hunk 2]
```diff
@@ -35,38 +36,59 @@ const ignoredFileExtensions = [".css", ".js", ".json"];
...
+      copyFileIfAbsentOrNewer(file, targetFile, logger);
+    });
   }
-  copyThemeFiles(themeFolder, staticAssetsThemeFolder, logger);
 }
 
 /**
- * Recursively copy files found in theme folder excluding any with a extension found in the `ignoredFileExtensions` array.
+ * Collect all folders with copyable files and all files to be copied.
+ * Foled will not be added if no files in folder or subfolders.
```
[the remark this answers is not in the document]
Comment (caalador) [reply flow-server/src/main/resources/plugins/application-theme-plugin/theme-copy.js]: fixed

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-copy.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pleku) [reply flow-server/src/main/resources/plugins/application-theme-plugin/theme-copy.js:30]: Right. I think I was thinking about `@StyleSheet` but confused it to a newer thing. And it is documented anyway now to be _external_ so it is not an issue

**Review by pleku — APPROVED** [review]

**Pull request merged**